src/NHibernate/Loader/Loader.cs

```diff
@@ -1533,17 +1533,17 @@ private IList ListIgnoreQueryCache(ISessionImplementor session, QueryParameters
 			return GetResultList(DoList(session, queryParameters), queryParameters.ResultTransformer);
 		}
 
-		private IList ListUsingQueryCache(ISessionImplementor session, QueryParameters queryParameters, ISet<string> querySpaces, IType[] resultTypes)
+		private IList ListUsingQueryCache(ISessionImplementor session, QueryParameters queryParameters, ISet<string> querySpaces, ref IType[] resultTypes)
 		{
 			IQueryCache queryCache = _factory.GetQueryCache(queryParameters.CacheRegion);
 
 			QueryKey key = GenerateQueryKey(session, queryParameters);
 
-			IList result = GetResultFromQueryCache(session, queryParameters, querySpaces, resultTypes, queryCache, key);
+			IList result = GetResultFromQueryCache(session, queryParameters, querySpaces, ref resultTypes, queryCache, key);
 
 			if (result == null)
 			{
-				result = DoList(session, queryParameters, key.ResultTransformer);
+				result = DoList(session, queryParameters, queryParameters.ResultTransformer);
```

Comment:
Why this change?

Reply:
Hi

This change is to be reverted, was inside for testing purposes.

Thx

LG